Formula/xcodegen.rb

```diff
@@ -0,0 +1,32 @@
+class Xcodegen < Formula
+  desc "Generate your Xcode project from a spec file and your folder structure"
+  homepage "https://github.com/yonaskolb/XcodeGen"
+  url "https://github.com/yonaskolb/XcodeGen/archive/1.10.3.tar.gz"
+  sha256 "30944e90b9855ff893c6581e0d6dcdf7e8ab2c0c5edadacba7fb5ce27164975a"
+
+  depends_on :xcode => ["9.3", :build]
+
+  def install
+    # fixes an issue an issue in homebrew when both Xcode 9.3+ and command line tools are installed
+    # see more details here https://github.com/Homebrew/brew/pull/4147
+    # remove this once homebrew is patched
+    ENV["CC"] = Utils.popen_read("xcrun -find clang").chomp
+
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    (testpath/"xcodegen.yml").write <<~EOS
+      name: GeneratedProject
+      options:
+        bundleIdPrefix: com.project
+      targets:
+        TestProject:
+          type: application
+          platform: iOS
+          sources: TestProject
+    EOS
+    (testpath/"TestProject").mkpath
+    system bin/"XcodeGen", "--spec", testpath/"xcodegen.yml"
```

Comment:
Can we assert something about what exists and the content of what exists after this step? See `mkcert` for instance:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/mkcert.rb#L29-L34

Reply:
Is that necessary? Xcodegen will fail if reading, parsing, generating or writing fails. The fact that it got through all those things is a valid test in my opinion. 
Validating the actuall structure of the project should be left to the actual xcodegen tests

Reply:
yes, it should assert something other than just 0 exit code

Reply:
Ok, I'll see if I can add something. Is there a way to run the tests locally? I couldn't find the command for that

Reply:
`brew test --verbose --debug xcodegen`

Reply:
Yeah I tried that but get a `Error: xcodegen is not linked` error

Reply:
```
brew link xcodegen
```